Using abs instead of std::abs in Exponent.cpp

See the [most recent workflow run](https://github.com/matthew-mccall/Oasis/actions/runs/6553213812/job/17798123609) for building Oasis. During the `Build` phase, the following warning is raised:

```
/home/runner/work/Oasis/Oasis/src/Exponent.cpp:63:51: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
   63 |                 Multiply<Expression> { Real { pow(abs(ImgCase->GetMostSigOp().GetMostSigOp().GetValue()), 0.5) },
      |                                                   ^
/home/runner/work/Oasis/Oasis/src/Exponent.cpp:63:51: note: use function 'std::abs' instead
   63 |                 Multiply<Expression> { Real { pow(abs(ImgCase->GetMostSigOp().GetMostSigOp().GetValue()), 0.5) },
      |                                                   ^~~
      |                                                   std::abs
```

--- a/src/Exponent.cpp
+++ b/src/Exponent.cpp
@@ -60,7 +60,7 @@ auto Exponent<Expression>::Simplify() const -> std::unique_ptr<Expression>
     if (auto ImgCase = Exponent<Multiply<Real, Expression>, Real>::Specialize(simplifiedExponent); ImgCase != nullptr) {
         if (ImgCase->GetMostSigOp().GetMostSigOp().GetValue() < 0 && ImgCase->GetLeastSigOp().GetValue() == 0.5) {
             return std::make_unique<Multiply<Expression>>(
-                Multiply<Expression> { Real { pow(abs(ImgCase->GetMostSigOp().GetMostSigOp().GetValue()), 0.5) },
+                Multiply<Expression> { Real { pow(std::abs(ImgCase->GetMostSigOp().GetMostSigOp().GetValue()), 0.5) },
                     Exponent<Expression> { ImgCase->GetMostSigOp().GetLeastSigOp(), Real { 0.5 } } },
                 Imaginary {});
         }

